Report submodule

EmeraldE11 opened the pull request:
Implemented Soti-Reporting submodule properly, using code from the README file.

EmeraldE11 committed:
add SOTI Reporting as a submodule

EmeraldE11 committed:
troubleshooting to try and get the submodule properly set up.

EmeraldE11 committed:
fixing up submodule placement and naming

EmeraldE11 committed:
Cleanup temp files

EmeraldE11 committed:
Attempt to reset branch

EmeraldE11 committed:
Attempt to reset branch, and added gitignore to remove temp files.

EmeraldE11 committed:
Fixed git change window error

EmeraldE11 committed:
Implemented Soti-Reporting submodule.

Added some code from the Soti-Reporting README.md to properly implement the crash log handling, specifically the WriteUnhandledExceptionToFile() method. I also tested with controlled crashes in previous commits to ensure functionality.

ziptyler commented on Soti-Reporting:
I made one more fix to the Soti-Reporting repo that you'll need to pull into this repo. Just update it to the latest commit on master.

ziptyler commented on WpfHelloWorld/MainWindow.xaml.cs:
```diff
@@ -89,5 +95,13 @@ private void HideButton_Click(object sender, RoutedEventArgs e)
+        // For unhandled UI exceptions.
+        private void WriteUnhandledExceptionToFile(object sender, UnhandledExceptionEventArgs args)
+        {
+            string appName = Assembly.GetExecutingAssembly().GetName().Name;
```
This line causes the VS compiler to generate a warning. I also see three "Messages" in the VS Error List related to different lines in MainWindow.xaml.cs. Please eliminate all warnings and messages, and then resubmit this PR.

ziptyler reviewed (changes requested):
Almost there. See comments for a few minor updates.

EmeraldE11 committed:
Pulled in the newest version of Soti-Reporting.

EmeraldE11 committed:
Code cleanup

Added a null check to the WriteUnhandledExceptionToFile, shortened the "new" expression to the Bitmap and OpenFileDialog, and added the dialog's filter in its initialization.

ziptyler reviewed (approved):
Looks good!
